kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/visitor/linux/IfcfgConfigWriter.java

```diff
@@ -181,6 +180,9 @@ private void writeRedhatConfig(NetInterfaceConfig<? extends NetInterfaceAddressC
                                             .append(((NetConfigIP4) netConfig).getGateway().getHostAddress())
                                             .append("\n");
                                 }
+                            } else {
```

Comment:
@cdealti I think we need to discuss this further ... On the one hand, setting either interface 'Status' or/and 'Configuration' to 'Unmanaged' implies that this interface will not be touched by Kura at all so that another network manager can handle it. And there may be network managers that are fully capable to configure ethernet and wireless interfaces. 
On the other hand, c5921 expects the following:
1. Interfaces controlled by c5921 (i.e. eth, wlan) are configured in the 'manual' (i.e. BOOTPROTO=none) mode in Linux but there is nothing within c5921 to do it. So it is desirable for Kura to be able configure those interfaces in 'manual' mode all the way to the /etc/network/interfaces file.
2. Wireless interface can't be configured by c5921, so that Kura needs to be able to configure hostapd and wpa_supplicant using the 'Wireless' tab.
3. We may also want to set NAT rules (but not DHCP server) in Linux for interfaces controlled by c5921. So that having NAT tab for "unmanaged" interface may be desirable. For example I could not make PPP work w/o masquerading. 

Let's discuss this further tomorrow ...

Reply:
@ibinshtok now I see what you mean!

However, manual mode may require more than setting e.g. ``iface eth1 inet manual`` (on Debian systems) so I'm not sure it's really worth in general, for example if one wants this in its /etc/network/interfaces:

```
auto eth0
iface eth0 inet manual
       up ifconfig $IFACE 0.0.0.0 up
       up /usr/local/bin/myconfigscript
       down ifconfig $IFACE down
```

would this PR allow this by setting the interface mode to "manual"?

Wouldn't be better to just not manage the interface configuration at all from ESF (Unmanaged) and ask the user to edit /etc/network/interfaces to add the manual configuration or whatever they like.

For production, you can seed a custom /etc/network/interfaces file and snapshot_0.xml as a post install phase.

Reply:
@cdealti No, my PR would not be able to handle advanced entries in the /etc/network/interfaces file. And you are right, those entries may be desired. So, if we leave the /etc/network/interfaces file alone, I think it would be the best to add the 'Unmanaged' option to interface 'Status' field only. So that if 'Status' is set to 'Unmanaged' all other fields in the 'TCP/IP' configuration tabs are disabled. In this case, interface are not going to be managed at all, except ensuring that it is up after 'Status' is switched to 'Unmanaged'. Do you agree?

Reply:
@cdealti Another question ... What are we going to do with 'Wireless' and 'DHCP & NAT' tabs if interface 'Status' is set to 'Unmanaged'? 

For the 'Wireless' tab:
Since IOS provides no way to configure hostapd and wpa_supplicant, it would be nice to allow Kura to do this if Kura's capabilities to configure wireless are sufficient. In other words, if all that customers want is WPA/WPA2 PSK, they can use Kura. If they need something like EAP-TLS, they need to handle it outside Kura. With this in mind, I recommend to add 'Unmanaged' option to the 'Wireless Mode' configuration field of the 'Wireless' tab. Do you agree with this?

For the 'DHCP & NAT' tab:
If the interface mode is set to 'Unmanaged', I would keep this tab enabled (at least for now) but default settings for the 'Router Mode' option would be 'Off'.